### PR TITLE
Enable content library problems for metrics tab

### PR DESCRIPTION
**Pull request by dcadams**
Problems imported from a content library have a different xmodule structure than regular problems. Changes were necessary to include these types.

### CODE REVIEW

**Comment by dcadams** [comment]
@kluo

**Comment by dcadams** [comment]
retest this please

File: lms/djangoapps/class_dashboard/dashboard_data.py [hunk 1]
```diff
@@ -249,6 +249,65 @@ def get_problem_set_grade_distrib(course_id, problem_set, enrollment):
     return prob_grade_distrib
 
 
+def construct_problem_data(prob_grade_distrib, total_student_count, c_subsection, c_unit, c_problem, component):
```
Comment (kluo) [inline lms/djangoapps/class_dashboard/dashboard_data.py]: Since this is just a refactoring, can we keep the whitespace as before so it's more cleanly organized?

File: lms/djangoapps/class_dashboard/dashboard_data.py [hunk 2]
```diff
@@ -530,9 +568,14 @@ def get_array_section_has_problem(course_id):
         for subsection in section.get_children():
             for unit in subsection.get_children():
                 for child in unit.get_children():
-                    if child.location.category == 'problem':
+                    if child.location.category in PROB_TYPE_LIST:
```
Comment (kluo) [inline lms/djangoapps/class_dashboard/dashboard_data.py]: PROB_TYPE_LIST includes 'lti', is this intended? Was the old check intentionally excluding 'lti' type problems?

File: lms/djangoapps/class_dashboard/tests/test_dashboard_data.py [hunk 3]
```diff
@@ -98,7 +111,7 @@ def setUp(self):
             for j, user in enumerate(self.users):
                 StudentModuleFactory.create(
                     grade=1 if i < j else 0,
-                    max_grade=1 if i < j else 0.5,
+                    max_grade=1,
```
Comment (kluo) [inline lms/djangoapps/class_dashboard/tests/test_dashboard_data.py]: Why is this change needed?

File: lms/djangoapps/class_dashboard/dashboard_data.py (hunk 2 above)
Comment (dcadams) [inline lms/djangoapps/class_dashboard/dashboard_data.py]: Excluding lti here was a bug.

File: lms/djangoapps/class_dashboard/tests/test_dashboard_data.py (hunk 3 above)
Comment (dcadams) [inline lms/djangoapps/class_dashboard/tests/test_dashboard_data.py]: Now I've added a content lib problem the number of 'problems' doesn't match the number of users. The test wasn't written that well several years ago.

**Comment by kluo** [comment]
+1, can you do a squash before merging, thanks.

**Comment by caesar2164** [comment]
👍
